### DISCUSSION TITLE
Is SSR can Run in localhost ? (Next 12)

### DISCUSSION THREAD

**Discussion by thanhtutzaw** [vercel/next.js#42873] [Help]
Code is right but SSR doesn't work in localhost. But some times it works in local. But In production (netlify) It works every single time .No fail . No error . I run in my local with different browser and It still shows error. Cleaning Browser Storage also didn't helped !
### I use these things

1. nextjs 12 & SSR
2. next-firebase-auth
3. firestore , cookies , firbase & auth

### Posted issues in

1. https://github.com/gladly-team/next-firebase-auth/issues/580#issue-1446747852
2. https://github.com/gladly-team/next-firebase-auth/discussions/572#discussion-4541226

**Comment by zakcodez** [comment]
Try next 13, or maybe try your private (network only) IP address

**Comment by icyJoseph** [comment]
Hi,

What SSR are you using? GetServerSideProps? I would test with Caddy reverse proxy to get HTTPS out of the box. My thesis being that some feature you are using requires HTTPS protocol, and locally you are running on HTTP.


However I would like to know more about what do you mean with:

- no fail no error?
- ssr? be more specific
- how are cookies being set?
- are you running `next build && next start` locally?

**Comment by thanhtutzaw** [comment]
I means :

- My app works in https (neltify)
- I use ssr getServersideProps with locahost ( sometimes it works ) but after I added ssr for multiple pages (localhost:3000/id1) I got error
- Cookies set by next-firebase-auth with firebase config
- I already tried with both next dev and next build 

Now I will test with Caddy Reverse Proxy Thanks

**Reply by icyJoseph** [reply]
Ok, if that doesn't work, I'd like to see a repository, it'll save us a lot of time :)

**Reply by thanhtutzaw** [reply]
Ok Sir please Give your Twitter or Telegram

**Reply by icyJoseph** [reply]
No worries, I think I found your repository. What error are you actually seeing? That's still not clear to me. 

Just to be clear, **_SSR does work locally_**, so it must something specific to your setup. 

What's the error you see? do you have any logs? screenshot?

**Reply by thanhtutzaw** [reply]
Ssr is work sometimes in localhost. But After I added new SSR for view page , cannot run SSR again in my localhost.

### Error Messages

1. login api status 500 (in browser)
2. Firebase Admin ( cannot create custom token ) and uid empty error.

**Reply by thanhtutzaw** [reply]
in local
1. SSR works
2. Sometimes it doesn't work them I clean browser storage,remove .next and run again. ### It works again !

3.Then Added Another SSR for  view pages
4. Now ,  project that use next-firbase-auth are all stop working. Got same 2 errors as I mentioned above. ( Not only my project, all project with next firebase auth)
